### PR TITLE
libraries/web-extensions: List objects automatically in Makefile.

### PR DESCRIPTION
**Pull request by Ambrevar**

### CODE REVIEW

**Comment by Ambrevar** [comment]
Actually, what if you moved `-fPIC -shared` to the LDFLAGS, and then remove the `libnyxt.so` rule, Make should handle the rest automatically via the default rules, nah?

**Comment by aartaka** [comment]
> Actually, what if you moved `-fPIC -shared` to the LDFLAGS, and then remove the `libnyxt.so` rule, Make should handle the rest automatically via the default rules, nah?

I don't see an implicit rule for .so files in [Implicit Rules Catalogue](https://www.gnu.org/software/make/manual/html_node/Catalogue-of-Rules.html#Catalogue-of-Rules)...

File: libraries/web-extensions/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ CC := ${shell command -v ${CC} 2> /dev/null || echo gcc}
 CFLAGS += ${shell pkg-config ${PKG_MODULES} --cflags} -fPIC
 LDLIBS += ${shell pkg-config ${PKG_MODULES} --libs}
 
-objects = nyxt.o globals.o extevent.o browser.o tabs.o management.o runtime.o extension.o storage.o alarms.o bookmarks.o browser_action.o commands.o history.o notifications.o permissions.o web_navigation.o web_request.o
+objects = $(patsubst %.c,%.o,$(wildcard *.c))
```
Comment (aartaka) [inline libraries/web-extensions/Makefile:13]: The rest of the Makefile uses `${}` syntax, using `$()` is inconsistent :)

**Comment by aartaka** [comment]
Yes, it does work!

**Comment by aartaka** [comment]
I mean that this `$(patsubst %.c,%.o,$(wildcard *.c))` spell works.

File: libraries/web-extensions/Makefile (hunk 1 above)
Comment (aartaka) [inline libraries/web-extensions/Makefile:13]: Should `objects` be capitalized? What are the conventions for Makefile variable naming, actually?

Comment (Ambrevar) [reply libraries/web-extensions/Makefile:13]: Sorry about that,  but actually this makefile deviates from the syntax of nyxt/Makefile :)

Comment (Ambrevar) [reply libraries/web-extensions/Makefile:13]: I'm unsure, but I believe uppercase is for overridable environment variables, so keeping this lowercase is the right thing to do as far as I can tell.

**Comment by Ambrevar** [comment]
> I don't see an implicit rule for .so files in [Implicit Rules Catalogue](https://www.gnu.org/software/make/manual/html_node/Catalogue-of-Rules.html#Catalogue-of-Rules)... 

```sh
$ make -p | grep '.so'
make: *** No targets specified and no makefile found.  Stop.
# This is free software: you are free to change and redistribute it.
...
.LIBPATTERNS = lib%.so lib%.a
```

There is .LIBPATTERNS, but I don't know how it's used, so I guess it
won't work.

OK to merge then?

File: libraries/web-extensions/Makefile (hunk 1 above)
Comment (aartaka) [reply libraries/web-extensions/Makefile:13]: Yes, that's sane :)

**Comment by aartaka** [comment]
OK to merge!

File: libraries/web-extensions/Makefile (hunk 1 above)
Comment (aartaka) [reply libraries/web-extensions/Makefile:13]: I'll change it after we merge then xD

**Pull request merged**